src/mds/nameserver2/namespace_service.cpp

```diff
@@ -1484,15 +1483,19 @@ void NameSpaceService::OpenFile(::google::protobuf::RpcController* controller,
         }
         return;
     }
-
-    ProtoSession *protoSession = new ProtoSession();
     FileInfo *fileInfo = new FileInfo();
+    ProtoSession *protoSession = new ProtoSession();
     CloneSourceSegment* cloneSourceSegment = new CloneSourceSegment();
+    uint64_t permission = 0;    // default permission is reader
     retCode = kCurveFS.OpenFile(request->filename(),
-                                clientIP,
+                                request->clientip(),
                                 protoSession,
                                 fileInfo,
-                                cloneSourceSegment);
+                                cloneSourceSegment,
+                                request->clientport(),
+                                &permission,
+                                request->date(),
+                                request->tobewriter());
```

Comment:
consider packing these parameters as a struct, like, `OpenContext`, long parameter list is a bad practice.